src/sentry/deletions/defaults/group.py

```diff
@@ -1,6 +1,59 @@
 from __future__ import absolute_import, print_function
 
-from ..base import ModelDeletionTask, ModelRelation
+from sentry import eventstore, nodestore
+from sentry.models import Event
+
+from ..base import BaseDeletionTask, BaseRelation, ModelDeletionTask, ModelRelation
+
+
+class GroupNodeDeletionTask(BaseDeletionTask):
+    """
+    Deletes nodestore data for group
+    """
+
+    DEFAULT_CHUNK_SIZE = 10000
+
+    def __init__(self, manager, group_id, project_id, **kwargs):
+        self.group_id = group_id
+        self.project_id = project_id
+        self.last_event = None
+        super(GroupNodeDeletionTask, self).__init__(manager, **kwargs)
+
+    def chunk(self):
+        conditions = []
+        if self.last_event is not None:
+            conditions.extend(
+                [
+                    ["timestamp", "<=", self.last_event.timestamp],
+                    [
+                        ["timestamp", "<", self.last_event.timestamp],
+                        ["event_id", "<", self.last_event.event_id],
+                    ],
+                ]
+            )
+
+        events = eventstore.get_events(
+            filter=eventstore.Filter(
+                conditions=conditions, project_ids=[self.project_id], group_ids=[self.group_id]
+            ),
+            limit=self.DEFAULT_CHUNK_SIZE,
+            referrer="deletions.group",
+            orderby=["-timestamp", "-event_id"],
+        )
```

Comment:
Are we guaranteed that this operation happens before clickhouse has dropped the events for retention ? What about failures ? How does the retry policy works and is the deletion retried before the events are dropped by snuba ?

Reply:
I believe nodestore values will expire after the retention period, so we only need to cover the case where groups are deleted via a user action prior to the retention period end.